protoc-gen-grpc-gateway/gengateway/template.go

```diff
@@ -0,0 +1,198 @@
+package gengateway
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+
+	"github.com/gengo/grpc-gateway/protoc-gen-grpc-gateway/descriptor"
+)
+
+type param struct {
+	*descriptor.File
+	Imports []descriptor.GoPackage
+}
+
+func applyTemplate(p param) (string, error) {
+	w := bytes.NewBuffer(nil)
+	if err := headerTemplate.Execute(w, p); err != nil {
+		return "", err
+	}
+	var methodSeen bool
+	for _, svc := range p.Services {
+		for _, meth := range svc.Methods {
+			methodSeen = true
+			if err := handlerTemplate.Execute(w, meth); err != nil {
+				return "", err
+			}
+		}
+	}
+	if !methodSeen {
+		return "", errNoTargetService
+	}
+	if err := trailerTemplate.Execute(w, p.Services); err != nil {
+		return "", err
+	}
+	return w.String(), nil
+}
+
+var (
+	headerTemplate = template.Must(template.New("header").Parse(`
+// Code generated by protoc-gen-grpc-gateway
+// source: {{.GetName}}
+// DO NOT EDIT!
+
+/*
+Package {{.GoPkg.Name}} is a reverse proxy.
+
+It translates gRPC into RESTful JSON APIs.
+*/
+package {{.GoPkg.Name}}
+import (
+	{{range $i := .Imports}}{{if $i.Standard}}{{$i | printf "%s\n"}}{{end}}{{end}}
+
+	{{range $i := .Imports}}{{if not $i.Standard}}{{$i | printf "%s\n"}}{{end}}{{end}}
+)
+
+var _ codes.Code
+var _ io.Reader
+var _ = runtime.String
```

Comment:
You may want to insert `var _ = json.Marshal` or similar here, otherwise it will potentially generate Go code that does not compile:

```
....pb.gw.go:13: imported and not used: "encoding/json"
```